pkg/model/utxo/milestone_diff.go

```diff
@@ -93,6 +113,37 @@ func (ms *MilestoneDiff) kvStorableLoad(utxoManager *Manager, key []byte, value
 		spents[i] = spent
 	}
 
+	hasTreasury, err := marshalUtil.ReadBool()
+	if err != nil {
+		return err
+	}
+
+	if hasTreasury {
+		treasuryOutputMilestoneID, err := marshalUtil.ReadBytes(iotago.MilestoneIDLength)
+		if err != nil {
+			return err
+		}
+
+		treasuryOutput, err := utxoManager.readUnspentTreasuryOutputWithoutLocking(treasuryOutputMilestoneID)
```

Comment:
Very good catch! This would have surely ramified itself into a panic if a snapshot would have been produced of which ms diffs contained would have spent treasury outputs.